### ISSUE TITLE
Prerelease versions are not detected as version changes

### ISSUE THREAD

**Issue by larskinn** [EndBug/version-check#11]
**Describe the bug**
Bumping a package from e.g, `2.0.5` to `3.0.0-beta.0` is not detected as a version change

**To Reproduce**
- Have a package with a non-prerelease version number (e.g. `2.0.5`)
- Run `npm version premajor --preid=beta`
- Push to the branch configured to trigger the `version-check` action

**Expected behavior**
`version-check` should detect a new version

**Additional context**
I think there may be two unrelated issues with detecting prerelease versions:
 1. `npm version` starts the first prerelease number at `.0`, but there was a bug in `semver-regex` which only supported `.1` and upwards (see https://github.com/sindresorhus/semver-regex/issues/18). The latest version of `version-check` was released on January 22, while the `semver-regex` fix was released on January 25.

2. Even `3.0.0-beta.1` is not recognized as a new version by `version-check`. Maybe it has something to do with the fact that `matchVersion('3.0.0-beta.1')` returns `'3.0.0'`?

    ```js
    (('3.0.0-beta.1'.match(/[0-9.]+/g) || [])
        .map(s => s.match(/(?<=^v?|\sv?)(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(?:0|[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*)(?:\.(?:0|[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*))*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?(?=$|\s)/ig))
        .find(e => !!e) || [])[0]
    
    => "3.0.0"
    ```
    If I understand [this line in main.ts correctly](https://github.com/EndBug/version-check/blob/master/src/main.ts#L195), `versions.added` will be `'3.0.0'`, and is compared to `'3.0.0-beta.1'` from the version field in `package.json` .

**Comment by EndBug** [comment]
I think this is my fault: I put `.match(/[0-9.]+/g)` in order to only select versions, but that cuts out other words like `-beta.0`. I'm fixing this asap, I just need to use something like `.split(' ')`

**Comment by EndBug** [comment]
Hey, this should be fixed by [`v1.2.1`](https://github.com/EndBug/version-check/releases/tag/v1.2.1) :)

**Comment by EndBug** [comment]
@all-contributors please add @larskinn for bug
